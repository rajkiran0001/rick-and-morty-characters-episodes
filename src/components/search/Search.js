import React, {useState, useEffect} from 'react';
import {Link} from '@reach/router';
import axios from 'axios';
import '../../App.css';

function Search() {
    const [data, setData] = useState({});
    const [query, setQuery] = useState('');
    const [search, setSearch] = useState('');
    const [loading, setLoading] = useState(true);

    const handleSubmit = e => {
        e.preventDefault();
        setSearch(query);
    };

    const searchCharacter = async () => {
        console.log('searching...');
        let queryString = '';
        const page = 25

        if (search) {
            queryString = `?name=${search}`;
        } else {
            queryString = `?page=${page}`;
        }

        const result = await axios(
            `https://rickandmortyapi.com/api/character/${queryString}`,
        );

        setData(result.data);
        setLoading(false);
    };

    useEffect(
        () => {
            searchCharacter();
        },
        [search],
    );

    return loading ? (
        <p>Loading...</p>
    ) : (
        <div className="results-wrapper">
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    placeholder="rick, morty, jerry, etc"
                    value={query}
                    onChange={e => setQuery(e.target.value)}
                />
                <button type="submit">Search</button>
            </form>
            <div className="results">
                <ul>
                    {data.results.map(item => (
                        <li key={item.id}>
                            <Link to={`/details/${item.id}`}>
                                <img alt={item.name} src={item.image} />
                                <span className="name"> {item.name}</span>
                            </Link>
                        </li>
                    ))}
                </ul>
            </div>
        </div>
    );
}

export default Search;
