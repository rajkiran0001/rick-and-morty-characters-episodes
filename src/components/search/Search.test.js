import React from "react";
import { shallow, mount } from "enzyme";
import Search from "./Search";
import Spinner from '../../Spinner';

describe("Search Page Snapshot ", () => {
  let mountWrapper;
  beforeEach(() => {
      mountWrapper = mount(<Search />);
  })

  afterEach(() => {
      mountWrapper.unmount();
  })

  it("renders Correctly", () => {
      expect(mountWrapper).toMatchSnapshot();
  })
  
});

describe("Search Page rendering of element", () => {
  let wrapper;
  beforeEach(() => {
      wrapper = shallow(<Search />);
  })

  afterEach(() => {
      wrapper.unmount();
  })
  
  it("Renders Spinner React Compount", ()=> {
      expect(wrapper.find(Spinner).render())
  })

  it("loop though the map function", () => {
    const wrapper = shallow(<Search />);
    wrapper.find(".name").forEach((node) => {
      expect(node.hasClass("name")).to.equal(true);
    });
  });

});
