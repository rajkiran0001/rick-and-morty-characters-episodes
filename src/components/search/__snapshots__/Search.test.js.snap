// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Product Page Snapshot  renders Correctly 1`] = `ReactWrapper {}`;

exports[`Search Page Snapshot  renders Correctly 1`] = `ReactWrapper {}`;
